lib/util/code-level-metrics.js

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const logger = require('../logger').child({ component: 'code-level-metrics' })
+
+/**
+ * Helper to retrieve Code Level Metrics(CLM)
+ * properties from a function reference.
+ *
+ * @param {Function} fn function reference
+ * @returns {object} CLM properties
+ */
+module.exports = function getCLMMeta(fn) {
```

Comment:
Not to be a buzzkill, but the agent-spec has more rules that I don't see accounted for here, specifically 

```
Agents MUST:

drop the attribute if it's over 255 characters
if code.function doesn't exist or if the function name is over 255 characters, drop the CLM attributes
if code.function exists and both code.namespace and code.filepath are over 255 characters, drop the CLM attributes
```

Reply:
ah good catch. I was going to do this then go distracted with finding an easier solution for retrieving the metadata

Reply:
this is built into our addAttribute, so eventually when these keys get associated with a span it'll drop if > 255 https://github.com/newrelic/node-newrelic/blob/main/lib/attributes.js#L127

Reply:
actually we truncate values, so i will build this to align with spec